src/services/refactors/convertParamsToDestructuredObject.ts

```diff
@@ -99,13 +115,38 @@ namespace ts.refactor.convertParamsToDestructuredObject {
             const functionSymbols = map(functionNames, getSymbolTargetAtLocation);
             const classSymbols = map(classNames, getSymbolTargetAtLocation);
             const isConstructor = isConstructorDeclaration(functionDeclaration);
+            const interfaceSymbols = map(functionNames, getSymbolForInterfaceSignature);
 
             for (const entry of referenceEntries) {
-                if (entry.kind !== FindAllReferences.EntryKind.Node) {
+                if (entry.kind === FindAllReferences.EntryKind.Span) {
                     groupedReferences.valid = false;
                     continue;
                 }
 
+                /* Calls and declarations of implemented interface methods should point to the symbol of the method signature */
+                if (contains(interfaceSymbols, getSymbolTargetAtLocation(entry.node))) {
+                    if (isValidMethodSignature(entry.node.parent)) {
+                        groupedReferences.signature = entry.node.parent;
+                        continue;
+                    }
+                    const call = entryToFunctionCall(entry);
+                    if (call) {
+                        groupedReferences.functionCalls.push(call);
+                        continue;
+                    }
+                }
+
+                /* Declarations in interface implementations have their own symbol so we need to go to the interface declaration
+                to get the type from the method signature. */

```

Comment:
I don't understand which part gets the type from the method signature. Is it `entryToDeclaration`? `getSymbolForInterfaceSignature` only gets the symbol.

Reply:
changed comment.